packages/a-msgpack/src/ExtData.ts

```diff
@@ -1,8 +1,10 @@
+export const WRITE_ONLY_HEADER = 'WRITE_ONLY_HEADER';
+
 /**
  * ExtData is used to handle Extension Types that are not registered to ExtensionCodec.
  */
 export class ExtData {
-  constructor(readonly type: number, readonly data: Uint8Array) {
+  constructor(readonly type: number, readonly data: Uint8Array | typeof WRITE_ONLY_HEADER) {
```

Comment:
data.length == 0 wouldn't work because data can be undefined/null? what if you make your have a constant of `Uint8Array` of size 0 to signify no data? Then the argument can only be of the same type and you'd only have to check if payload is zero to not write it as well as use its size in header directly rather than having and if statement.

Reply:
Hmm yeah this is a good idea. Let me try that